ports/rp2/rp2_pio.c

```diff
@@ -357,6 +381,30 @@ static mp_obj_t rp2_pio_state_machine(size_t n_args, const mp_obj_t *pos_args, m
 }
 MP_DEFINE_CONST_FUN_OBJ_KW(rp2_pio_state_machine_obj, 2, rp2_pio_state_machine);
 
+#if PICO_PIO_VERSION > 0
+// PIO.gpio_base(0|16)
+static mp_obj_t rp2_pio_gpio_base(mp_obj_t self_in, mp_obj_t gpio_base_in) {
+    rp2_pio_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    uint8_t gpio_base = mp_obj_get_int(gpio_base_in);
+
+    // TODO check for RP2350B?
```

Comment:
Aha, `gpio_base` is only applicable on RP2350B where there are 48 IOs instead of 32. It can be 0 or 16 to give the PIO access to pins 0-31 or 16-47 respectively.

As such this function should do something reasonable, like raise a ValueError on RP2040 or RP235XA hardware.

We should probably also check the value is one of the two valid inputs, but it started to feel like a lot of input validation for a currently slightly obscure (albeit Pimoroni have shipped a Pico-a-like with a B variant chip) method.

Reply:
I've now removed this todo, and made this method available only on RP2350B.  It's a pretty special method and must be called before any PIO programs are added (or StateMachine's created).  I've also added docs for this new `PIO.gpio_base()` method to describe its use.